server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -1998,6 +2018,9 @@ public synchronized void setSchemaTemplate(SetTemplatePlan plan) throws Metadata
 
       templateManager.checkIsTemplateAndMNodeCompatible(template, node);
 
+      // node might be replaced when check with alignment
+      node = mtree.checkTemplateAlignmentWithMountedNode(node, template);
+
       node.setSchemaTemplate(template);
```

Comment:
If this node is not using template, why should it be replaced by entityMNode?

Reply:
That's true. This check should be a part of setUsingTemplate methods rather than this one.